lender.hpp

```diff
@@ -196,10 +202,41 @@ class library : public recycle_memory<T> {
     auto operate() -> buffer_ptr<T>;
 };
 
+template <typename T>
+class mailbox : public recycle_memory<T> {
+    friend class mail_test;
+
+    private:
+        int max_read;
+        // TODO: does map_value need to be a template or not?
+        // template <typename T>

```

Comment:
I was under the impression that because `map_value` is within `private` of `mailbox`, the static creation example you have wouldn't work. If that is not the case, I would assume that `protected` does not help in preventing errant `map_value`s from being constructed?

Generally I'm not sure anyone would find the map_value useful outside of this specific context but it's not particularly harmful if a user wants to instantiate it themselves I suppose...